src/test/java/uk/gov/pay/connector/charge/service/ChargeExpiryServiceTest.java

```diff
@@ -176,24 +182,60 @@ public void shouldExpireChargesWithStatus_awaitingCaptureRequest_byCallingProvid
-                    ChargeEntity chargeEntity = ChargeEntityFixture.aValidChargeEntity()
-                            .withAmount(200L)
-                            .withCreatedDate(ZonedDateTime.now())
-                            .withStatus(status)
-                            .withGatewayAccountEntity(gatewayAccount)
-                            .build();
-                    chargeExpiryService.expire(singletonList(chargeEntity));
-
-                    try {
-                        verify(mockPaymentProvider, never()).cancel(any());
-                    } catch (GatewayException ignored) {}
-
-                    verify(mockChargeService).transitionChargeState(chargeEntity.getExternalId(), EXPIRED);
-                });
+    @Parameters({
+            "CREATED",
+            "ENTERING CARD DETAILS",
+            "AUTHORISATION 3DS REQUIRED",
+            "AUTHORISATION 3DS READY"
+    })
+    public void shouldExpireChargesWithoutCallingProviderToCancel(String chargeStatus) throws Exception {
+        var status = ChargeStatus.fromString(chargeStatus);
+        ChargeEntity chargeEntity = ChargeEntityFixture.aValidChargeEntity()
+                .withAmount(200L)
+                .withCreatedDate(ZonedDateTime.now())
+                .withStatus(status)
+                .withGatewayAccountEntity(gatewayAccount)
+                .build();
+
+        when(mockQueryService.isTerminableWithGateway(chargeEntity)).thenReturn(false);
+
+        chargeExpiryService.expire(singletonList(chargeEntity));
+
+        verify(mockPaymentProvider, never()).cancel(any());
+        verify(mockChargeService).transitionChargeState(chargeEntity.getExternalId(), EXPIRED);
+    }
+
+    @Test
+    @Parameters({
+            "AUTHORISATION 3DS REQUIRED",
+            "AUTHORISATION 3DS READY"
+    })
+    public void shouldExpireWithPaymentProvider_whenGatewayStateIsCancellable(String chargeStatus) throws Exception {
+        var status = ChargeStatus.fromString(chargeStatus);
+        
+        ChargeEntity chargeEntity = ChargeEntityFixture.aValidChargeEntity()
+                .withAmount(200L)
+                .withCreatedDate(ZonedDateTime.now())
+                .withStatus(status)
+                .withGatewayAccountEntity(gatewayAccount)
+                .build();
+
+        when(mockQueryService.isTerminableWithGateway(chargeEntity)).thenReturn(true);
+
+        when(mockWorldpayCancelResponse.cancelStatus()).thenReturn(CancelStatus.CANCELLED);
+
+        when(mockChargeDao.findByExternalId(chargeEntity.getExternalId())).thenReturn(Optional.of(chargeEntity));
+        when(mockPaymentProvider.cancel(any())).thenReturn(gatewayResponse);
+        ArgumentCaptor<CancelGatewayRequest> cancelCaptor = ArgumentCaptor.forClass(CancelGatewayRequest.class);
+
+        ChargeEntity expiredCharge = mockExpiredChargeEntity();
+        when(mockChargeService.transitionChargeState(any(String.class), any())).thenReturn(expiredCharge);
```

Comment:
This is how I originally did this, but I think we can make it better
```suggestion
        when(mockChargeService.transitionChargeState(chargeEntity.getExternalId(), EXPIRED)).thenReturn(expiredCharge);
```

Reply:
Done